File: Serverside/Admin/Api/Dtos/Variant/VariantDto.cs
using FNO.Admin.AdminApi.Dtos.Product;
using System.Collections.Generic;

namespace FNO.Admin.AdminApi.Dtos.Variant
{
    public class VariantDto : BaseVariantDto
    {
        public ProductDto Product { get; set; }
        public List<InventoryInfoDto> InventoryInfos { get; set; }
    }
}

File: Clientside/Shop/src/models/VariantDto.ts
import BaseVariantDto from "./BaseVariantDto";
import ColorDto from "./ColorDto";
import InventoryInfoDto from "./InventoryInfoDto";
import ProductDto from "./ProductDto";

interface VariantDto extends BaseVariantDto {
    product: ProductDto
    inventoryInfos: InventoryInfoDto[]
}

export default VariantDto